.github/workflows/set-automatic-deploys-enabled.yaml

```diff
@@ -0,0 +1,115 @@
+# USAGE
+# -----
+#
+# name: Set automatic_deploys_enabled (optionally image_tag too)
+#
+# on:
+#   workflow_dispatch:
+#     inputs:
+#       resetImageTag:
+#         description: 'Whether to reset image tag to main'
+#         required: false
+#         default: false
+#         type: boolean
+#       automaticDeploysEnabled:
+#         description: 'Set whether automatic deploys is active or not'
+#         required: false
+#         default: true
+#         type: boolean
+#       environment:
+#         description: 'Environment to deploy to'
+#         required: true
+#         type: choice
+#         options:
+#         - integration
+#         - staging
+#         - production
+#         default: 'integration'
+#
+# jobs:
+#   set_automatic_deploys_enabled:
+#     name: Set automatic_deploys_enabled to ${{ github.event.inputs.environment }}
+#     uses: alphagov/govuk-infrastructure/.github/workflows/automatic_deploys_enabled.yaml@main
+#     with:
+#       resetImageTag: ${{ github.event.inputs.environment.resetImageTag }}
+#       automaticDeploysEnabled: ${{ github.event.inputs.environment.automaticDeploysEnabled }}
+#       environment: ${{ github.event.inputs.environment || 'integration' }}
+#     secrets:
+#       WEBHOOK_TOKEN: ${{ secrets.ARGO_EVENTS_WEBHOOK_TOKEN }}
+#       WEBHOOK_URL: ${{ secrets.GOVUK_INTEGRATION_ARGO_EVENTS_WEBHOOK_URL }}
+#       GOVUK_CI_GITHUB_API_TOKEN: ${{ secrets.GOVUK_CI_GITHUB_API_TOKEN }}
+
+name: Set automatic_deploys_enabled (optionally image_tag too)
+
+on:
+  workflow_call:
+    inputs:
+      resetImageTag:
+        description: 'Whether to reset image tag to main'
+        required: false
+        default: false
+        type: boolean
+      automaticDeploysEnabled:
+        description: 'Set whether automatic deploys is active or not'
+        required: false
```

Comment:
Would it make sense for this to be required? (given that the purpose of the job is to set it one way or the other? Or are they other usability implications of it being a required input?)

Similar question for `environment` below also.

Reply:
I guess if i put required true, we don't need the default?

Reply:
Anyway sorry didn't mean to split hairs - I should have prefixed this with "optional"! By all means try it out either way and as long as the UI isn't super awkward it's all good with me :)

Reply:
No worries, I was thinking about it before but could not see the required info. :-)